NameIt

```diff
@@ -4,24 +4,28 @@ import os
 import re
 import string
 import fitz  # PyMuPDF
+import unicodedata
+import subprocess
 from habanero import Crossref
 
+#checking for internet connection
+def check_internet_access():
+    try:
+        # Run the ping command to check internet connectivity
+        subprocess.check_call(["ping", "-c", "1", "google.com"])
+        return True
+    except subprocess.CalledProcessError:
+        return False
+    
 # limiting filenames to valid character
 def remove_invalid_characters(text):
     valid_characters = "-_.() %s%s" % (string.ascii_letters, string.digits)
-    cleaned_text = ''.join(c for c in text if c in valid_characters)
+    cleaned_text = ''.join(
+        c if c in valid_characters or unicodedata.category(c) in ('Mn', 'Mc', 'Ll', 'Lu', 'Lt', 'Lo') else '_'
+        for c in text
+    )
     return cleaned_text
 
-# truncating to lmit the file name for system to manage
```

Comment:
a different form of truncation was used to shorten the title and this code is added unnecessary complication to the code.